theano/sandbox/cuda/basic_ops.py

```diff
@@ -2445,6 +2445,8 @@ class GpuReshape(tensor.Reshape, GpuOp):
 
     # __hash__, __eq__, __str__ come from tensor.Subtensor
     def make_node(self, x, shp):
+        x = as_cuda_ndarray_variable(x)
+        shp = tensor.as_tensor_variable(shp)
```

Comment:
@abergeron Should we rather wrap x and shp inside the test? The only reason I could see for GpuReshape.make_node not being shielded against host/device variables is because GpuReshape is not in the public interface.